distributed/rpc/batch/parameter_server.py

```diff
@@ -0,0 +1,149 @@
+import os
+import threading
+from datetime import datetime
+
+import torch
+import torch.distributed.rpc as rpc
+import torch.multiprocessing as mp
+import torch.nn as nn
+from torch import optim
+
+import torchvision
+
+
+batch_size = 20
+image_w = 64
+image_h = 64
+num_classes = 30
+batch_update_size = 5
+
+
+@rpc.functions.async_execution
+def _update_and_fetch_model(ps_rref, grads):
+    return ps_rref.local_value().update_and_fetch_model(grads)
+
+
+def timed_log(text):
+    print(f"{datetime.now().strftime('%H:%M:%S')} {text}")
+
+
+class BatchUpdateParameterServer(object):
+
+    def __init__(self, batch_update_size=batch_update_size):
+        self.model = torchvision.models.resnet50(num_classes=num_classes)
+        self.lock = threading.Lock()
+        self.future_model = torch.futures.Future()
+        self.batch_update_size = batch_update_size
+        self.curr_update_size = 0
+        self.optimizer = optim.SGD(self.model.parameters(), lr=0.001, momentum=0.9)
+        for p in self.model.parameters():
+            p.grad = torch.zeros_like(p)
```

Comment:
Is `p.grad.zero_()` more canonical?

Reply:
`p.grad` is actually `None` initially. `self.optimizer.zero_grad()` also won't touch None grads.